server/server.go

```diff
@@ -140,11 +140,12 @@ func NewServer(config *Config) (*Server, error) {
 		return nil, fmt.Errorf("failed to create data directories: %w", err)
 	}
 
-	if err := m.setupTelemetry(); err != nil {
-		return nil, err
-	}
-
 	if config.Telemetry.PrometheusAddr != nil {
+		// Only setup telemetry if `PrometheusAddr` has been configured.
+		if err := m.setupTelemetry(); err != nil {
```

Comment:
Not sure is this change safe, because `setupTelemetry` initializes global telemetry object, which is latter used when reporting metrics:
- https://github.com/0xPolygon/polygon-edge/blob/develop/txpool/txpool.go#L277,
- https://github.com/0xPolygon/polygon-edge/blob/develop/network/server.go#L517,
- https://github.com/0xPolygon/polygon-edge/blob/develop/network/server.go#L772 etc. 
Since we don't check is `PrometheusAddr` provided when we report metrics from different modules, I believe it would fail whenever we try to report it (because as I've mentioned previously, we wouldn't have global metrics object initialized).

So to me, if we are going to use this approach, we would need to check when we are reporting metrics is `PrometheusAddr` provided in order not to cause `panic`.

Reply:
https://github.com/hashicorp/go-metrics/blob/v0.4.1/start.go#L46-L49